implementations/python/sugar-lift-python-source/src/sugar_lift_python_source/dependency_export_adapter.py

```
@@ -198,13 +236,39 @@ def _resolve_export_uncached(
             session=session,
         )
     if binding is not None and binding[0] == "alias":
-        return resolve_export(
+        # Follow the binding of the RHS name that *reaches* the assignment —
+        # never the RHS name's final module binding after later reassignment.
+        assign_node, name_node = binding[1]
+        alias_name = name_node.id
+        hop_key = (module_name, alias_name)
+        if hop_key in seen:
+            return _gap(
```

Ruff (0.15.21)

[error] Undefined name `_gap`

(F821)